src/remote_attestation.md

```diff
@@ -1,57 +1,56 @@
 # Remote Attestation
 
-TODO
-
 The goal of attestation is to convince a third party that a specific piece of code is running on a genuine Intel SGX HW.
 
-## convincing the Integritee user
+## Convincing the Integritee user
 
-A user that interacts with Integritee wants to be sure that the shielding pubkey she uses to encrypt her call to the STF originates form an enclave that
+A user that interacts with Integritee wants to execute a state update without sharing her data and therefore encrypt her call to the STF. She has to be sure that the shielding public key she uses for encryption originates from an enclave that
+<!-- TODO: define enclave before using -->
 
-   1. is running on genuine Intel SGX HW
-   1. runs the official code
-   1. accesses the correct STF state
+   1. is running on genuine Intel SGX HW,
+   1. runs the official code and
+   1. accesses the correct STF state.
 
 ### Classical Remote Attestation Solution
 
-The standard use case for remote attestation involves a service provider (SP) like a video streaming service which wants to be sure his viewer application runs on a genuine SGX HW and respects DRM.
-The SP therefore requests a quote from the application and sends that quote to Intel Attestation Services (IAS) who sign off the quote if it is genuine.
+The standard use case for remote attestation (RA) involves a service provider (SP) like a video streaming service that wants to be sure that his viewer application runs on a genuine SGX HW and respects DRM.
+The SP therefore requests a quote from the application and sends that quote to the Intel Attestation Services (IAS) who signs off the quote if it is genuine. This process certifies the enclave to be genuine and the SP can safely run his calls. 
+
+However, that IAS only talks to registered clients. You need to register in order to get a SPID which needs to be supplied along with requests.
 
-The issue here is that IAS only talks to registered clients. You need to register in order to get a SPID which needs to be supplied along with requests.
+![Attestation of the enclave](./fig/attestation.svg)
 
 ### Attestation Registry On-Chain
 
-It isn't practical to ask every client to register with Intel and perform RA before every request. Therefore we'd rather let the integritee-worker operators attest their enclaves with IAS and write the signed quote and their certificate to the blockchain for everyone to verify.
+It isn't practical to ask every client to register with Intel and perform RA before every request. An integritee-worker operator attests his own enclave with IAS. The signed quote and the certificate is then written onto the blockchain and everyone can verify the genuineness of the enclave.
```

Comment:
the certificate isn't just "written" to the blockchain! It is verified by the blockchain (the certificate is part of an extrinsic, but it never becomes part of the state)

Reply:
I only modified minor words in this paragraph.